### PR TITLE
[Azure AD roles] Restricted management administrative unit properties (Beta)

### PR DESCRIPTION
**Pull request by rolyon**
This docs PR is tracked via this work item https://dev.azure.com/msft-skilling/Content/_workitems/edit/54353/?triage=true

### CODE REVIEW

File: api-reference/beta/resources/device.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ This resource lets you add your own data to custom properties using [extensions]
 |id|String|The unique identifier for the device. Inherited from [directoryObject](directoryobject.md). Key, Not nullable. Read-only. Supports `$filter` (`eq`, `ne`, `not`, `in`). |
 |isCompliant|Boolean|`true` if the device complies with Mobile Device Management (MDM) policies; otherwise, `false`. Read-only. This can only be updated by Intune for any device OS type or by an [approved MDM app](/windows/client-management/mdm/azure-active-directory-integration-with-mdm) for Windows OS devices. Supports `$filter` (`eq`, `ne`, `not`).|
 |isManaged|Boolean|`true` if the device is managed by a Mobile Device Management (MDM) app; otherwise, `false`. This can only be updated by Intune for any device OS type or by an [approved MDM app](/windows/client-management/mdm/azure-active-directory-integration-with-mdm) for Windows OS devices. Supports `$filter` (`eq`, `ne`, `not`). |
+|isManagementRestricted|Boolean|`true` if the device should be treated as sensitive, which requires specific permissions to manage. Default value is `false`. Read-only.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/device.md:67]: The API.md isn't clear on what these "special permissions" are. Kindly clarify/document if there are restrictions. For example, if this device isn't part of a restricted AU but is a restricted device, who can manage it? Is it still only Global Administrator and Privileged Role Administrator?

File: api-reference/beta/resources/group.md [hunk 2]
```diff
@@ -139,6 +139,7 @@ This resource supports:
...
+| isManagementRestricted         | Boolean                                                                  | Indicates whether the group should be treated as sensitive, which requires specific permissions to manage. Default value is `false`. Read-only.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/group.md:142]: Same comment in the device mention

File: api-reference/beta/resources/user.md [hunk 3]
```diff
@@ -197,6 +197,7 @@ This resource supports:
 | imAddresses | String collection | The instant message voice over IP (VOIP) session initiation protocol (SIP) addresses for the user. Read-only. Supports `$filter` (`eq`, `not`, `ge`, `le`, `startsWith`).|
 | infoCatalogs | String collection | Identifies the info segments assigned to the user.  Supports `$filter` (`eq`, `not`, `ge`, `le`, `startsWith`). |
 | interests | String collection | A list for the user to describe their interests. <br><br>Returned only on `$select`. |
+| isManagementRestricted| Boolean | `true` if the user should be treated as sensitive, which requires specific permissions to manage. Default value is `false`. Read-only. |
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/user.md:200]: Same comment in the device mention

File: api-reference/beta/api/user-list.md [hunk 4]
```diff
@@ -835,6 +835,42 @@ Content-type: application/json
...
 
+### Example 13: List all users whose management is restricted
+
+#### Request
+<!-- {
+  "blockType": "request",
+  "name": "get_user_restricted"
+}-->
+```http
+GET https://graph.microsoft.com/beta/users?$filter=isManagementRestricted eq true&$select=displayName,userPrincipalName
```
Comment (FaithOmbongi) [inline api-reference/beta/api/user-list.md:846]: Q: Do you need special permissions/roles to successfully make this request?

File: api-reference/beta/api/group-get.md [hunk 5]
```diff
@@ -295,6 +295,33 @@ Content-type: application/json
...
 
+### Example 4: Check if management of a group is restricted
+
+#### Request
+<!-- {
+  "blockType": "request",
+  "name": "get_group_restricted"
+}-->
+```http
+GET https://graph.microsoft.com/beta/groups/aed0b780-965f-4149-85c5-a8c73e58brt6?$select=isManagementRestricted
```
Comment (FaithOmbongi) [inline api-reference/beta/api/group-get.md:306]: Q: Do you need special permissions/roles to successfully make this request?

File: api-reference/beta/api/directory-post-administrativeunits.md [hunk 6]
```diff
@@ -117,6 +115,47 @@ Content-type: application/json
 }
 ```
 
+### Example 2: Create a restricted management administrative unit
```
Comment (FaithOmbongi) [inline api-reference/beta/api/directory-post-administrativeunits.md:118]: I think it's a "restricted administrative unit" meaning its management is restricted - not a "restricted management administrative unit"

Let's call out the special role requirements
```suggestion
### Example 2: Create a restricted administrative unit
```

File: api-reference/beta/api/administrativeunit-update.md [hunk 7]
```diff
@@ -97,7 +101,34 @@ Content-type: application/json
...
+<!-- {
+  "blockType": "response"
+} -->
+```http
+HTTP/1.1 204 No Content
+```
+
+### Example 2: Update a restricted management administrative unit
+
+The following example changes a restricted management administrative unit to a regular administrative unit.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/administrativeunit-update.md:115]: Call out the role requirements

File: api-reference/beta/api/administrativeunit-get.md [hunk 8]
```diff
@@ -111,6 +113,38 @@ Content-type: application/json
...
 
+### Example 2: Get a restricted management administrative unit
+
+#### Request
+<!-- {
+  "blockType": "request",
+  "name": "get_administrativeunit_restricted"
+}-->
+```http
+GET https://graph.microsoft.com/beta/administrativeUnits/2sd35b05-ae71-48ab-9e7d-4r41a28te37d
```
Comment (FaithOmbongi) [inline api-reference/beta/api/administrativeunit-get.md:124]: Any role requirements for READ operations?

File: changelog/Microsoft.DirectoryServices.json [hunk 9]
```diff
@@ -18,6 +18,48 @@
...
       "SubArea": ""
     },
+    {
+      "ChangeList": [
+        { 
+          "Id": "12f096a3-6bd8-49db-8753-24c9272a288b",
+          "ApiChange": "Property",
+          "ChangedApiName": "IsMemberManagementRestricted",
+          "ChangeType": "Addition",
+          "Description": "Added the **IsMemberManagementRestricted** property to the [administrativeUnit](https://docs.microsoft.com/en-us/graph/api/resources/administrativeunit?view=graph-rest-beta) resource.",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:28]: Suggestion to enhance the description. Please tweak where applicable.
```suggestion
          "Description": "Added the **isMemberManagementRestricted** property to the [administrativeUnit](https://docs.microsoft.com/en-us/graph/api/resources/administrativeunit?view=graph-rest-beta) resource. Use this property to define administrative units whose management is restricted to a limited group of administrators with the appropriate permissions and specific Azure AD administrative roles.",
```

File: changelog/Microsoft.DirectoryServices.json [hunk 10]
```diff
@@ -18,6 +18,48 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **IsManagementRestricted** property to the [group](https://docs.microsoft.com/en-us/graph/api/resources/group?view=graph-rest-beta) resource.",
+          "Target": "group"
+        },
+        { 
+          "Id": "12f096a3-6bd8-49db-8753-24c9272a288b",
+          "ApiChange": "Property",
+          "ChangedApiName": "IsManagementRestricted",
+          "ChangeType": "Addition",
+          "Description": "Added the **IsManagementRestricted** property to the [user](https://docs.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-beta) resource.",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.DirectoryServices.json:52]: Suggestions to enhance the descriptions.
```suggestion
          "Description": "Added the **isManagementRestricted** property to the [device](https://docs.microsoft.com/en-us/graph/api/resources/device?view=graph-rest-beta) resource. Use this property to define devices whose management is restricted to a limited group of administrators with the appropriate permissions and specific Azure AD administrative roles.",
          "Target": "device"
        },
        { 
          "Id": "12f096a3-6bd8-49db-8753-24c9272a288b",
          "ApiChange": "Property",
          "ChangedApiName": "IsManagementRestricted",
          "ChangeType": "Addition",
          "Description": "Added the **isManagementRestricted** property to the [group](https://docs.microsoft.com/en-us/graph/api/resources/group?view=graph-rest-beta) resource. Use this property to define groups whose management is restricted to a limited group of administrators with the appropriate permissions and specific Azure AD administrative roles.",
          "Target": "group"
        },
        { 
          "Id": "12f096a3-6bd8-49db-8753-24c9272a288b",
          "ApiChange": "Property",
          "ChangedApiName": "IsManagementRestricted",
          "ChangeType": "Addition",
          "Description": "Added the **isManagementRestricted** property to the [user](https://docs.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-beta) resource. Use this property to define users whose profile management is restricted to a limited group of administrators with the appropriate permissions and specific Azure AD administrative roles.",
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @#for your work on this PR. 

- I've left suggestions to call out the role requirements where they apply and a couple to enhance the changelog descriptions.
- It seems from the schema that the isManagementRestricted property is also going live for applications and service principals. Can you confirm and include these in your PR?

A point of note though: In the [administrative units docs](https://docs.microsoft.com/en-us/graph/api/resources/administrativeunit?view=graph-rest-beta), we don't mention that applications and service principals can be members of administrative units. If apps and service principals should be included in the list of possible members that can be added, retrieved, or removed from AUs, let's close that gap too.

File: api-reference/beta/api/administrativeunit-get.md (hunk 8 above)
Comment (KaTabish) [reply api-reference/beta/api/administrativeunit-get.md:124]: No. Default user should be able to read.

File: api-reference/beta/api/administrativeunit-update.md (hunk 7 above)
Comment (KaTabish) [reply api-reference/beta/api/administrativeunit-update.md:115]: Only Global Admins and Privileged Role admins are allowed to perform write operations on AUs.
This applies to all AU create/update/delete operations.

File: api-reference/beta/api/directory-post-administrativeunits.md (hunk 6 above)
Comment (KaTabish) [reply api-reference/beta/api/directory-post-administrativeunits.md:118]: It is indeed called a "restricted management admin unit". The management of the AU is not restricted, but the management of the individual members of the AU is restricted.
No special role requirement other than what applies to AUs in general.

File: api-reference/beta/api/group-get.md (hunk 5 above)
Comment (KaTabish) [reply api-reference/beta/api/group-get.md:306]: No. Default user can read this.

File: api-reference/beta/api/user-list.md (hunk 4 above)
Comment (KaTabish) [reply api-reference/beta/api/user-list.md:846]: No

File: api-reference/beta/resources/administrativeunit.md [hunk 11]
```diff
@@ -61,6 +61,7 @@ This topic provides descriptions of the declared properties and navigation prope
 |description|String|An optional description for the administrative unit. Supports `$filter` (`eq`, `ne`, `in`, `startsWith`), `$search`.|
 |displayName|String|Display name for the administrative unit. Supports `$filter` (`eq`, `ne`, `not`, `ge`, `le`, `in`, `startsWith`, and `eq` on `null` values), `$search`, and `$orderBy`.|
 |id|String|Unique identifier for the administrative unit. Read-only. Supports `$filter` (`eq`).|
+|isMemberManagementRestricted|Boolean|`true` if members of this administrative unit should be treated as sensitive, which requires specific permissions to manage. Default value is `false`.|
```
[the remark this answers is not in the document]
Comment (KaTabish) [reply api-reference/beta/resources/administrativeunit.md:64]: Not sure I understand: "If the caller lacks these roles but has the permissions"?
Default users can read this value, so you don't have to be a GA/PRA to see it.

File: api-reference/beta/resources/device.md (hunk 1 above)
Comment (KaTabish) [reply api-reference/beta/resources/device.md:67]: GA/PRA cannot manage restricted device. Also a device cannot be restricted if it isn't a member of a restricted AU.
I think we should change it to " 'true' if the device is a member of a restricted management admin unit, in which case it requires a role scoped to the restricted AU to manage "

File: api-reference/beta/resources/group.md (hunk 2 above)
Comment (KaTabish) [reply api-reference/beta/resources/group.md:142]: see above

File: api-reference/beta/resources/user.md (hunk 3 above)
Comment (KaTabish) [reply api-reference/beta/resources/user.md:200]: see above

File: changelog/Microsoft.DirectoryServices.json (hunk 10 above)
Comment (KaTabish) [reply changelog/Microsoft.DirectoryServices.json:52]: I don't think we should say "use this property to define" as it cannot be directly controlled by end user.
We can use a common definition like "This property denotes objects whose management is restricted to admins scoped over restricted management admin units the object is a member of".

File: changelog/Microsoft.DirectoryServices.json (hunk 9 above)
Comment (KaTabish) [reply changelog/Microsoft.DirectoryServices.json:28]: ....to define admins units whose roles don't inherit from tenant level administrators, and management of individual member objects is limited to admins scoped over a restricted management admin unit.

File: api-reference/beta/api/administrativeunit-update.md (hunk 7 above)
Comment (rolyon) [reply api-reference/beta/api/administrativeunit-update.md:120]: Prerequisite roles were added in a separate pull request:
https://github.com/microsoftgraph/microsoft-graph-docs/pull/17106

File: changelog/Microsoft.DirectoryServices.json (hunk 9 above)
Comment (rolyon) [reply changelog/Microsoft.DirectoryServices.json:224]: I dont think we should add specific info in the changelog that could out of date. Please see updated description in administrativeunit.md

File: changelog/Microsoft.DirectoryServices.json (hunk 10 above)
Comment (rolyon) [reply changelog/Microsoft.DirectoryServices.json:248]: I dont think we should add specific info in the changelog that could out of date. Please see updated descriptions in device.md, group.md, and user.md

**Comment by rolyon** [comment]
Hi @FaithOmbongi 
Can you review changes?
thanks

File: api-reference/beta/resources/group.md [hunk 12]
```diff
@@ -141,6 +141,7 @@ This resource supports:
...
+| isManagementRestricted         | Boolean                                                                  | Indicates whether the group is a member of a restricted management administrative unit, in which case it requires a role scoped to the restricted administrative unit to manage. Default value is `false`. Read-only.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/group.md:144]: SUGGESTION: Global for other resources that have this property - I got from Kamran's response that the AU isn't restricted - but it's management of the members of the AU that's restricted.
```suggestion
| isManagementRestricted         | Boolean                                                                  | Indicates whether the group is a member of a restricted management administrative unit, in which case it requires a role scoped to the administrative unit to manage. Default value is `false`. Read-only.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```

File: concepts/permissions-reference.md [hunk 13]
```diff
@@ -753,6 +753,7 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 | _Directory.Read.All_ |Read directory data | Allows the app to read data in your organization's directory, such as users, groups and apps. **Note**: Users may consent to applications that require this permission if the application is registered in their own organization’s tenant.| Yes | No |
 | _Directory.ReadWrite.All_ |Read and write directory data | Allows the app to read and write data in your organization's directory, such as users, and groups. It does not allow the app to delete users or groups, or reset user passwords. | Yes | No |
 | _Directory.AccessAsUser.All_ |Access directory as the signed-in user  | Allows the app to have the same access to information in the directory as the signed-in user. | Yes | No |
+| _Directory.Write.Restricted_ |Manage all restricted resources  | Allows an app to manage restricted resources as per existing permissions of the app, on behalf of the signed-in user. | Yes | No |
```
Comment (FaithOmbongi) [inline concepts/permissions-reference.md:756]: Align these with the official description that's surfaced to devs and admins.
image

```suggestion
| _Directory.Write.Restricted_ |Manage restricted resources in the directory  | Allows the app to manage restricted resources based on the other permissions granted to the app, on behalf of the signed-in user. | Yes | No |
```

File: concepts/permissions-reference.md [hunk 14]
```diff
@@ -762,6 +763,7 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 |:-----------------------------|:-----------------------------------------|:-----------------|:-----------------|
 | _Directory.Read.All_ | Read directory data | Allows the app to read data in your organization's directory, such as users, groups and apps, without a signed-in user. | Yes |
 | _Directory.ReadWrite.All_ | Read and write directory data | Allows the app to read and write data in your organization's directory, such as users, and groups, without a signed-in user. Does not allow user or group deletion. | Yes |
+| _Directory.Write.Restricted_ |Manage all restricted resources  | Allows an app to manage restricted resources as per existing permissions of the app, without a signed-in user. | Yes |
```
Comment (FaithOmbongi) [inline concepts/permissions-reference.md:766]:
```suggestion
| _Directory.Write.Restricted_ |Manage restricted resources in the directory  | Allows the app to manage restricted resources based on the other permissions granted to the app, without a signed-in user. | Yes |
```

**Review by FaithOmbongi — APPROVED** [review]
Hi @rolyon - I've left suggestions for minor fixes like using the proper property name casing + updates to the permissions descriptions. Otherwise, approved with suggestions.

File: api-reference/beta/resources/group.md (hunk 12 above)
Comment (KaTabish) [reply api-reference/beta/resources/group.md:144]: That's correct, but I think Robert has used "restricted" to refer to the type of AU (as the role has to be scoped to the RMAU and not just any AU).
Maybe reword as "in which case managing the group requires a role scoped to the restricted management admin unit"

**Comment by FaithOmbongi** [comment]
Hi @rolyon - If there's no reason we should hold back this docs PR, please get it into a mergeable state and replace the **do not merge** with the **ready to merge** label.

**Comment by rolyon** [comment]
Hi @FaithOmbongi
This feature is still in progress.

**Comment by Lauragra** [comment]
@rolyon , please resolve the merge conflicts before we merge this PR. I don't want to do it incorrectly. Thanks.

File: api-reference/beta/api/directory-post-administrativeunits.md [hunk 15]
```diff
@@ -61,11 +61,13 @@ Because the **administrativeUnit** resource supports [extensions](/graph/extensi
 
 If successful, this method returns a `201 Created` response code and an [administrativeUnit](../resources/administrativeunit.md) object in the response body.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/directory-post-administrativeunits.md:62]: Also update the "Request body" section above.

File: api-reference/beta/api/directory-post-administrativeunits.md [hunk 16]
```diff
@@ -61,11 +61,13 @@ Because the **administrativeUnit** resource supports [extensions](/graph/extensi
...
 
-## Example
+## Examples
 
-### Request
+### Example 1: Create an administrative unit
 
-The following is an example of a request that creates a new administrative unit with a dynamic membership rule to include all users whose country is United States.
+The following is an example creates a new administrative unit with a dynamic membership rule to include all users whose country is United States.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/directory-post-administrativeunits.md:68]:
```suggestion
The following example creates a new administrative unit with a dynamic membership rule to include all users whose country is United States.
```

File: api-reference/beta/api/directory-post-administrativeunits.md [hunk 17]
```diff
@@ -132,6 +134,49 @@ Content-type: application/json
 }
 ```
 
+### Example 2: Create a restricted management administrative unit
+
+The following creates a new restricted management administrative unit. The `isMemberManagementRestricted` property is immutable, so cannot be changed later.
```
Comment (FaithOmbongi) [inline api-reference/beta/api/directory-post-administrativeunits.md:139]:
```suggestion
The following example creates a new restricted management administrative unit. The `isMemberManagementRestricted` property is immutable, so cannot be changed later.
```

File: api-reference/beta/api/administrativeunit-update.md [hunk 18]
```diff
@@ -105,7 +108,34 @@ Content-type: application/json
...
+#### Response
+
+<!-- {
+  "blockType": "response"
+} -->
+```http
+HTTP/1.1 204 No Content
+```
+
+### Example 2: Update the display name of an administrative unit
```
Comment (FaithOmbongi) [inline api-reference/beta/api/administrativeunit-update.md:120]: You can update Example 1 to update both properties given they are simple types. Will save real estate space for other complex update operations in the future.

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Hi @rolyon -

This PR has been on hold for a long while so took me a while to refresh my memory. I've reviewed it based on your last changes.

- Q: Is isManagementRestricted for devices, users, and groups also immutable? We currently indicate they are read-only.
- For group, user, and device entities, also add the properties to the "Request body" sections of the Create operations

File: api-reference/beta/api/directory-post-administrativeunits.md (hunk 15 above)
Comment (rolyon) [reply api-reference/beta/api/directory-post-administrativeunits.md:63]: Made update. See https://github.com/microsoftgraph/microsoft-graph-docs/pull/16295/commits/9486830d99a1b2dc9a516ed1554d2550adab43b7

**Comment by rolyon** [comment]
Hi @FaithOmbongi 

> * Q: Is isManagementRestricted for devices, users, and groups also immutable? We currently indicate they are read-only.
> * For group, user, and device entities, also add the properties to the "Request body" sections of the Create operations

* Checked with PM. Not immutable and can be changed.
* For Create operations, you can't set property with API, but property gets set when member is added.

**Comment by FaithOmbongi** [comment]
> Hi @FaithOmbongi
> 
> > * Q: Is isManagementRestricted for devices, users, and groups also immutable? We currently indicate they are read-only.
> > * For group, user, and device entities, also add the properties to the "Request body" sections of the Create operations
> 
> * Checked with PM. Not immutable and can be changed.
> * For Create operations, you can't set property with API, but property gets set when member is added.

Ah, gotcha. Thanks for the clarification @rolyon

**Review by FaithOmbongi — APPROVED** [review]
LGTM, approved!
